Tests/MessagingPush/MessagingPushTest.swift

```diff
@@ -0,0 +1,156 @@
+@testable import CioTracking
+@testable import CioMessagingPush
+import Foundation
+import SharedTests
+import XCTest
+
+class MessagingPushTest: UnitTest {
+    private var mockCustomerIO: CustomerIO!
+    private var messagingPush: MessagingPush!
+
+    private var identifyRepositoryMock: IdentifyRepositoryMock!
+    
+    private var httpClientMock: HttpClientMock!
+
+    override func setUp() {
+        super.setUp()
+
+        httpClientMock = HttpClientMock()
+        identifyRepositoryMock = IdentifyRepositoryMock()
+
+        mockCustomerIO = CustomerIO(credentialsStore: nil, sdkConfig: SdkConfig(), identifyRepository: identifyRepositoryMock,
+                                keyValueStorage: nil)
+        
+        mockCustomerIO.setCredentials(siteId: String.random, apiKey: String.random, region: Region.EU)
+        
+        messagingPush = MessagingPush(customerIO: mockCustomerIO, httpClient: httpClientMock, keyValueStorage: DITracking.shared.keyValueStorage)
+    }
+
+    // MARK: registerDeviceToken
+
+    func test_registerDeviceToken_expectFailIfNotIdentified() {
+ 
+        let expect = expectation(description: "Expect to fail to register device token")
+        self.messagingPush.registerDeviceToken(deviceToken: Data(String.random.utf8)) { result in
+            guard case .failure(let error) = result else { return XCTFail() }
+            guard case .notInitialized = error else { return XCTFail() }
+            expect.fulfill()
+        }
+        
+        waitForExpectations()
+    }
+    
+    func test_registerDeviceToken_givenHttpSuccess_expectSaveExpectedData() {
+        
+        identifyRepositoryMock.setIdentifierClosure = { identifier in
+            self.identifyRepositoryMock.identifier = identifier
```

Comment:
I would avoid setting properties on mocks since each time that this is done, values like `mock.identifierCalled: Bool` is changed. it may give us false positive/negatives with our tests.

Reply:
changed up the way the test works to avoid it. also removed `setIdentifier`